### PR TITLE
Fixes #21026: condition from variable match does not work with node properties

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/21026

### CODE REVIEW

**Comment by Fdall** [comment]
PR updated with a new commit

File: tree/30_generic_methods/condition_from_string_match.cf [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+# The regex is singleline, case sensitive, culture invariant and implicitly enclosed by the start/end of string
+# delimiters `^` and `$`.
+# This method's reporting status will always be "success".
+#
+# ### Examples
+#
+# ```yaml
+# -name: Define is_matching_true
+#  method: condition_from_string_match
+#    condition: is_matching
```
Comment (amousset) [inline tree/30_generic_methods/condition_from_string_match.cf:37]:
```suggestion
#  params:
#    condition: is_matching
```

**Comment by Fdall** [comment]
PR updated with a new commit

**Review by peckpeck — APPROVED** [review]

**Comment by Fdall** [comment]
PR rebased

**Pull request merged**